File: Java/Medium/WrongAnswer/minimum-operations-to-halve-array-sum_4.java
class Solution {
    public int halveArray(int[] nums) {
        
        int n = nums.length;
        int sum = 0;
        for(int i = 0; i < n; i++) {
            sum += nums[i];
        }
        Arrays.sort(nums);
        int count = 0;
        int total = 0;
        for(int i = n - 1; i >= 0; i--) {
            total += nums[i];
            count++;
            if(total >= sum / 2) {
                break;
            }
        }
        return count;
    }
}

File: Rust/Medium/WrongAnswer/minimum-operations-to-halve-array-sum_4.rs
impl Solution {
    pub fn halve_array(nums: Vec<i32>) -> i32 {
        
        let mut nums = nums;
        nums.sort();
        let mut sum = nums.iter().sum::<i32>();
        let mut res = 0;
        while sum > 0 {
            let n = nums.pop().unwrap();
            sum -= n;
            res += 1;
            if sum * 2 < n {
                break;
            }
        }
        res
    }
}